Build everything internally & fix apache builds

notcarbide opened the pull request:
Title explains it all. :)

(This was sitting locally for some time now without a commit, so I'm just kinda submitting as-is, but I can also break up the patches if needed.)

notcarbide committed:
sh: build: Update to build within source dir & for apache

- Build everything internally by default.

- Add an option to include htaccess files for apache servers. (if
  not, created wondercms zip will not work by default on apache)

notcarbide committed:
Add gitignore

- Ignore default outputs from build script

NicolasCARPi commented:
Thanks I'll have a more thorough look later. As a quick thing that I spotted, use `$(pwd)` instead of ``pwd`` as it is POSIX. See https://stackoverflow.com/questions/9449778/what-is-the-benefit-of-using-instead-of-backticks-in-shell-scripts

notcarbide committed:
Lint for POSIX standard

- Using the latest from:
    https://github.com/koalaman/shellcheck

- Fix typo in comment

notcarbide commented:
Excuse the mess haha. Most of what I know about bash (and coding in general) was interpreted from XDA-Developers, so it's all very likely to be all over the place. POSIX probably isn't the only good practice I'm not following ;)

NicolasCARPi commented on build.sh:
```diff
@@ -3,26 +3,77 @@
+    echo "Build in $BUILD_DIR successful!"
+    echo "Output: $ZIP_FILE"
+else
+    echo "Build command failed!"
```
You should add an "exit 1" here

NicolasCARPi replied:
Instead of having REAL_OPTION, wouldn't it be better to check if ZIP_APPEND is empty?

NicolasCARPi commented on build.sh:
```diff
@@ -3,26 +3,77 @@
+
+# check if command for apache was used
+# and only include files for relevant versions
+if [ -n "$1" ]; then
```
this isn't necessary imho

NicolasCARPi commented on build.sh:
```diff
@@ -3,26 +3,77 @@
+    rm "$ZIP_FILE"
+fi
+# also attempt to make zip path if does not exist
+if [ ! -d "$ZIP_PATH" ]; then
```
this should be moved after ZIP_PATH declaration

notcarbide committed:
Improvements based on comments

-  See @NicolasCARPi comments on #2

NicolasCARPi commented:
Thank you!
